Shared/src/com/pacoapp/paco/shared/util/SearchUtil.java

```diff
@@ -0,0 +1,237 @@
+    Select selectStatement = (Select)getJsqlStatement(selectSql);
+    PlainSelect pl = (PlainSelect)selectStatement.getSelectBody();
+    List<String> qvList = Lists.newArrayList();
+    getQueriedValue(pl.getWhere(), colName, qvList);
+    return qvList;
+  }
+  
+  public static void getQueriedValue(Expression node, String columnName, List<String> queriedValueList) {  
+    
+    if(node==null)  
+     return;  
+    if(node instanceof Parenthesis){
+      node = ((Parenthesis) node).getExpression();
+    } 
+
+    if(node instanceof BinaryExpression){
+      Expression le  = ((BinaryExpression)node).getLeftExpression();
+      Expression re  = ((BinaryExpression)node).getRightExpression();
+      
+      if ((le.getClass().getName().contains("Column")) && le.toString().equalsIgnoreCase(columnName)){
+        queriedValueList.add(((BinaryExpression)node).getRightExpression().toString());
+      }else{
+        getQueriedValue(le, columnName, queriedValueList);
+        getQueriedValue(re, columnName, queriedValueList);
+        
+      }
+    }
+
+    if(node instanceof InExpression){
+      if ((((InExpression) node).getLeftExpression().getClass().getName().contains("Column"))&& node.toString().contains(columnName)){
+        String listWithParen = ((InExpression)node).getRightItemsList().toString().replace('(', ' ');
+        String listWithoutParen = listWithParen.replace(')', ' ' );
+        String[] arr = listWithoutParen.split(", ");
+        queriedValueList.addAll(Arrays.asList(arr));
+      }
+    }
+    if(node instanceof IsNullExpression){
+      if ((((IsNullExpression) node).getLeftExpression().getClass().getName().contains("Column"))&& node.toString().contains(columnName)){
+        queriedValueList.add("isnull");
+      }
+    }
+    
+  }    
+  
+  public static void getColumnNamesInWhere(Expression node, List<String> colNames){
+    getColumnNames(node, colNames);
+  }
+  
+  public static void getColumnNamesInHaving(Expression node, List<String> colNames){
+    getColumnNames(node, colNames);
+  }
+  
+  public static void getColumnNamesInSortOrder(String clause,  List<String> colNames){
+    getColumnNames(clause, colNames);
+  }
+  public static void getColumnNamesInGroupBy(String clause,  List<String> colNames){
+    getColumnNames(clause, colNames);
+  }
+  
+  private static net.sf.jsqlparser.statement.Statement getJsqlStatement(String selectSql){
+    net.sf.jsqlparser.statement.Statement statement = null;
+    try {
+      statement = CCJSqlParserUtil.parse(selectSql);
+    } catch (JSQLParserException e) {
+      // TODO Auto-generated catch block
+      e.printStackTrace();
+    }
+    List<String> colNames = Lists.newArrayList(); 
+    
+    Select selectStatement = (Select) statement;
+    return selectStatement;
+  }
+  
+  public static List<String> getAllColNamesInQuery(String selectSql){
+
+    List<String> colNames = Lists.newArrayList(); 
```

Comment:
Will try to incorporate jsql parser features(Column name finder, insert query) in the next update.